### PR TITLE
adding end to end build test for complex apt packages

### PR DESCRIPTION
**Pull request by sharifelgamal**

### CODE REVIEW

File: tests/package_managers/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BUILD.bazel
```
Comment (dlorenc) [inline tests/package_managers/.gitignore]: These are unusual entries in a git ignore, so could you comment explaining why?

File: tests/package_managers/test_complex_packages.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -ex
+# Build new BUILD file with download_pkgs target
+touch BUILD.bazel
```
Comment (dlorenc) [inline tests/package_managers/test_complex_packages.sh]: I don't think you need the touch here.

File: tests/package_managers/test_complex_packages.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+cat > Dockerfile.test <<- EOM
+FROM bazel/ubuntu:ubuntu_16_0_4_vanilla
+
+RUN apt-get update && \
+  apt-get install -y curl netbase ca-certificates
+EOM
+
+cid=$(docker build -q - < Dockerfile.test)
+
+docker save $cid > test_complex_pkgs.dockerfile.tar
```
Comment (dlorenc) [inline tests/package_managers/test_complex_packages.sh]: Container diff should be able to diff in the daemon -you can avoid this save.

File: tests/package_managers/.gitignore (hunk 1 above)
Comment (sharifelgamal) [reply tests/package_managers/.gitignore]: Done.

File: tests/package_managers/test_complex_packages.sh (hunk 2 above)
Comment (sharifelgamal) [reply tests/package_managers/test_complex_packages.sh]: Fixed.

File: tests/package_managers/test_complex_packages.sh (hunk 3 above)
Comment (sharifelgamal) [reply tests/package_managers/test_complex_packages.sh]: Yep. Fixed.

File: tests/package_managers/test_complex_packages.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+FROM bazel/ubuntu:ubuntu_16_0_4_vanilla
+
+RUN apt-get update && \
+  apt-get install -y curl netbase ca-certificates
+EOM
+
+cid=$(docker build -q - < Dockerfile.test)
+
+# Compare it with the tar file built with install_pkgs using container diff
+container-diff diff test_complex_install_pkgs.tar daemon://$cid
```
Comment (tejal29) [inline tests/package_managers/test_complex_packages.sh]: how to get this binary?
Is this present on travis?

File: tests/package_managers/test_complex_packages.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (tejal29) [inline tests/package_managers/test_complex_packages.sh]: Can this be run on travis? If not, can you add a todo?

File: tests/package_managers/test_complex_packages.sh (hunk 4 above)
Comment (sharifelgamal) [reply tests/package_managers/test_complex_packages.sh]: you can get container-diff through curl, documentation is at https://github.com/GoogleCloudPlatform/container-diff. I'm going to try to add this to travis now.

**Comment by dlorenc** [comment]
Can you try rebasing? I might have accidentally fixed this in #152

File: .travis.yml [hunk 6]
```diff
@@ -24,6 +24,7 @@ addons:
 
 install:
   - go get -u github.com/bazelbuild/buildifier/buildifier
+  - curl -LO https://storage.googleapis.com/container-diff/latest/container-diff-linux-amd64 && chmod +x container-diff-linux-amd64 && sudo mv container-diff-linux-amd64 /usr/local/bin/container-diff
```
Comment (tejal29) [inline .travis.yml]: Just a note we all need this on our machines something to mention in https://github.com/GoogleCloudPlatform/base-images-docker/blob/master/CONTRIB.md

File: package_managers/apt_get/apt_get.bzl [hunk 7]
```diff
@@ -77,6 +71,8 @@ def _impl(ctx):
         install_commands = install_commands,
     )
 
+    shell_file_contents.append('#!/bin/bash')
```
Comment (dlorenc) [inline package_managers/apt_get/apt_get.bzl]: Did you mean to make these changes?

**Comment by dlorenc** [comment]
Looks like travis is failing with:

```shell
dpkg: dependency problems prevent configuration of ca-certificates:
 ca-certificates depends on openssl (>= 1.0.0); however:
  Package openssl is not installed.
```

File: package_managers/apt_get/apt_get.bzl (hunk 7 above)
Comment (sharifelgamal) [reply package_managers/apt_get/apt_get.bzl]: Yes, this was generating malformed bash and breaking a couple of our bazel tests.

File: .travis.yml (hunk 6 above)
Comment (sharifelgamal) [reply .travis.yml]: Done.

**Pull request closed without merging**